CONTRIBUTING.md

```diff
@@ -42,14 +42,56 @@ Ensuring excellent documentation and thorough testing is absolutely crucial. Her
 - If you add a feature:
   - Include unit tests in the `test` directory. 
   - Add a demo script in the `examples` directory.
-
-- If you review the code:
-  - Ensure responsible code review principles and cautious code approval.
-  - Try to review not only the modified code snippets but also consider the overall impact and optimization of the PR from a global and system perspective.
-  - More information about code review in Github is [here](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/about-pull-request-reviews)
  
 We're a small team focused on building great things. If you have something in mind that you'd like to add or modify, opening a pull request is the ideal way to catch our attention. 🚀
 
+### Contributing to Code Reviews 🔍
+This part outlines the guidelines and best practices for conducting code reviews in CAMEL. The aim is to ensure that all contributions are of high quality, align with the project's goals, and are consistent with our coding standards.
+
+#### Purpose of Code Reviews
+- Maintain Code Quality: Ensure that the codebase remains clean, readable, and maintainable.
+- Knowledge Sharing: Facilitate knowledge sharing among contributors and help new contributors learn best practices.
+- Bug Prevention: Catch potential bugs and issues before they are merged into the main branch.
+- Consistency: Ensure consistency in style, design patterns, and architecture across the project.
+
+#### Review Process Overview
+- Reviewers should check the code for functionality, readability, consistency, and compliance with the project’s coding standards.
+- If changes are necessary, the reviewer should leave constructive feedback.
+- The contributor addresses feedback and updates the PR.
+- The reviewer re-reviews the updated code.
+- Once the code is approved by at least two reviewer, it can be merged into the main branch.
+- Merging should be done by a maintainer or an authorized contributor.
+
+#### Code Review Checklist
+- Functionality
+  - Correctness: Does the code perform the intended task? Are edge cases handled?
+  - Testing: Is there sufficient test coverage? Do all tests pass?
+  - Security: Are there any security vulnerabilities introduced by the change?
+  - Performance: Does the code introduce any performance regressions?
+
+- Code Quality
+  - Readability: Is the code easy to read and understand? Is it well-commented where necessary?
+  - Maintainability: Is the code structured in a way that makes future changes easy?
+  - Style: Does the code follow the project’s style guidelines?
```

Comment:
do we have a style guideline for now?

Reply:
I would suggest the Google style guides.(https://google.github.io/styleguide/pyguide.html) They are actively maintained. If we decide to use them, then all reviewers should get familiar with them. There may be some existing pre-commit tests regarding the styles, which we could leverage.

Reply:
For now we will do the style/format check using ruff, I will update the doc to make it more clear

Reply:
currently we are using ruff to format the code, I will update the doc to make it more clear

Reply:
I agree to keep using ruff as a code format and lint checker, but I think their checks are too rough for a PR, and passing the checks of ruff should be the prerequisite for raising a PR. When it comes to the reviewing of PR, we need a more specific standard, so that we can do faster and better, and make sure the codebase of CAMEL is consistent. (Also I think the contributors should also be aware of the code style we are using so maybe we can declare the code style in contribution guide)

Reply:
updated code style part

Reply:
Hi @Wendong-Fan. Our code style is a bit different from https://google.github.io/styleguide/pyguide.html. For example the dosctring is quite different. We may need to write it down with some examples.

Reply:
Hi @WHALEEYE. This is a good question. We should put the docstring in of `__init__` under the class instead of inside the method since this provide a better way for devlopers to use when they are using code editor like VS code.

More informative:
<img width="815" alt="image" src="https://github.com/user-attachments/assets/9c6e92be-5cd2-4204-bbae-8999afb51335">

Not informative:
<img width="317" alt="image" src="https://github.com/user-attachments/assets/71228221-c6c6-4487-b9f4-0335506192da">

There are many inconsistences in the code base. We should change them all to putting the docstring in of `__init__` under the class instead of inside the method. @Wendong-Fan @WHALEEYE

Reply:
Also we should use r-string and the default value we can write like: (default: :obj:`None`).

Reply:
I will create a new page to document a detailed review checklist. Over time, we can add items to this list, transforming it into a comprehensive guideline